src/redux/features/apiSlice.js

```diff
@@ -0,0 +1,47 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+export const apiSlice = createApi({
+  reducerPath: "api",
+  baseQuery: fetchBaseQuery({ baseUrl: "https://dummyjson.com/" }),
+  endpoints: (builder) => ({
+    getProducts: builder.query({
+      query: ({ page = 1, limit = 8 } = {}) =>
+        `products?limit=${limit}&skip=${(page - 1) * limit}`,
+    }),
+
+    getProduct: builder.query({
+      query: (id) => `products/${id}`,
+    }),
+
+    createProduct: builder.mutation({
+      query: (product) => ({
+        url: "products/add",
+        method: "POST",
+        body: product,
+      }),
+    }),
+
+    updateProduct: builder.mutation({
+      query: ({ id, ...updates }) => ({
+        url: `product/${id}`,
+        method: "PUT",
+        body: updates,
```

Comment:
updateProduct: builder.mutation({
  query: ({ id, ...updates }) => ({
    url: `product/${id}`,
    method: "PUT", // Change this to 'PATCH' for partial updates.
    body: updates,
  }),
}),